File: ROS_YuXiang/learn0_listen_topic/src/minimal_topic_py/minimal_topic_py/talker.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class Talker(Node):
    def __init__(self):
        super().__init__('talker')
        self.pub = self.create_publisher(String, 'hello', 10)
        self.timer = self.create_timer(1.0, self.timer_cb)

    def timer_cb(self):
        msg = String()
        msg.data = 'Hello, ROS 2 topic!'
        self.pub.publish(msg)
        self.get_logger().info(f'Published: {msg.data}')

def main():
    rclpy.init()
    rclpy.spin(Talker())
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: ROS_YuXiang/learn0_listen_topic/src/minimal_topic/src/talker.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);
  auto node = rclcpp::Node::make_shared("talker");
  auto pub  = node->create_publisher<std_msgs::msg::String>("hello", 10);
  rclcpp::Rate loop(1);          // 1 Hz

  while (rclcpp::ok()) {
    auto msg = std_msgs::msg::String();
    msg.data = "Hello, ROS 2 topic!";
    pub->publish(msg);
    RCLCPP_INFO(node->get_logger(), "Published: %s", msg.data.c_str());
    rclcpp::spin_some(node);
    loop.sleep();
  }
  rclcpp::shutdown();
  return 0;
}